index.js

```diff
@@ -58,10 +58,18 @@ class TemplateCompiler extends Filter {
   }
 
   processString(string, relativePath) {
-    return 'export default ' + utils.template(this.options.templateCompiler, stripBom(string), {
-      contents: string,
-      moduleName: relativePath
-    }) + ';';
+    try {
+      return 'export default ' + utils.template(this.options.templateCompiler, stripBom(string), {
+        contents: string,
+        moduleName: relativePath
+      }) + ';';
+    } catch(error) {
+      // augment with location and type information and re-throw.
+      error.type = 'Template Compiler Error';
+      error.location = error.location.start;
```

Comment:
@twokul ya, I would prefer if this was made more resilient as bit-rot or unexpected errors can always change what `catch` ends up catching. 

Its literally the worst to have a `catch` handler throw when a different bug happened. So if we can beef up this catch handler, it will help mitigate future sadness.